environment.yml

```diff
@@ -27,7 +27,7 @@ dependencies:
         - ipdb
         - ipywidgets
         - jsonmerge
-        - joblib==0.10.3
+        - joblib<0.13,>=0.12
```

Comment:
```(garage) garage (fix_sleeping_proc *) $ pip install joblib<0.13,>=0.12
bash: 0.13,: No such file or directory```

Reply:
try `pip install "joblib<0.13,>=0.12"`

Reply:
it works

Reply:
Is there a reason we're specifying a version here? The installed version (0.12.2) is equivalent to if I simply called `pip install joblib`.

Reply:
We are being more careful with the major changes introduced in joblib, so we are restricting the updates of this library to only minor changes.

Reply:
The semantic versioning scheme used by many open source libraries:
XX.YY.ZZ

XX changes -- no backwards compatibility. literally anything can happen. whole packages can disappear.
YY changes -- backwards compatible for the same value of YY. features may be added but not removed. most increments to YY will be small changes but they may sometimes be backwards-incompatible, especially for libraries <1.0.
ZZ changes -- bug fixes/maintenance within a release only. generally no new features.